### PR TITLE
devicestate,boot,tests: make `fakeinstaller` test work

### PR DESCRIPTION
**Pull request by mvo5**
Further adjustments and cleanups to ensure that the full `fakeinstaller` nested test that can boot into the generated image.

### CODE REVIEW

File: boot/makebootable.go [hunk 1]
```diff
@@ -327,21 +346,14 @@ func makeRunnableSystem(model *asserts.Model, bootWith *BootableSet, sealer *Tru
 	if err := os.MkdirAll(snapBlobDir, 0755); err != nil {
 		return err
 	}
-	for _, fn := range []string{bootWith.BasePath, bootWith.KernelPath, bootWith.GadgetPath} {
-		dst := filepath.Join(snapBlobDir, filepath.Base(fn))
```
Comment (mvo5) [inline boot/makebootable.go:331]: We probably don't need this change, `snap prepare-image` should put the right filename already into the seed

File: boot/makebootable.go [hunk 2]
```diff
@@ -312,6 +312,25 @@ type makeRunnableOptions struct {
 	AfterDataReset bool
 }
 
+func copyNoSymLink(orig, dst string) error {
+	// if the source path is a symlink, don't copy the symlink, copy the
+	// target file instead of copying the symlink, as the initramfs won't
+	// follow the symlink when it goes to mount the base and kernel snaps by
+	// design as the initramfs should only be using trusted things from
+	// ubuntu-data to boot in run mode
+	if osutil.IsSymlink(orig) {
```
Comment (mardy) [inline boot/makebootable.go:321]: Do you want this to be a `while` to be on the safe side?

File: overlord/devicestate/handlers_install.go [hunk 3]
```diff
@@ -1321,6 +1321,12 @@ func (m *DeviceManager) doInstallFinish(t *state.Task, _ *tomb.Tomb) error {
 		Gadget:     snapInfos[snap.TypeGadget],
 		GadgetPath: snapSeeds[snap.TypeGadget].Path,
 
+		// modeenv:recovery_system will be set based on
+		// RecoverySystemDir and seeding reads this var
+		// from modeenv to know what it's seeding from
+		RecoverySystemDir:   systemLabel,
+		RecoverySystemLabel: systemLabel,
```
Comment (pedronis) [inline overlord/devicestate/handlers_install.go:1328]: the code is not  very consistent about these, we probably want to make it work to just set RecoverySystemLabel

File: overlord/devicestate/devicemgr.go [hunk 4]
```diff
@@ -309,6 +309,7 @@ func (m *DeviceManager) StartUp() error {
 }
 
 func (m *DeviceManager) shouldMountUbuntuSave() bool {
+	// TODO: this should check DeviceCtx.IsClassicBoot() instead
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go:312]: I need to double check on this

File: boot/makebootable.go [hunk 5]
```diff
@@ -312,6 +312,25 @@ type makeRunnableOptions struct {
 	AfterDataReset bool
 }
 
+func copyNoSymLink(orig, dst string) error {
```
Comment (pedronis) [inline boot/makebootable.go:315]: this probably needs a different name/signature otherwise the comment in it is a bit odd

**Review by pedronis — COMMENTED** [review]
looking at this closer, this seems alright but it needs some tweaks, I will look into that

File: overlord/devicestate/devicemgr.go (hunk 4 above)
Comment (mvo5) [reply overlord/devicestate/devicemgr.go:312]: This was a bit of a drive-by comment, without something here the change 2  that creates the device key will fail with something like `"internal error: cannot access device keypair manager if ubuntu-save is unavailable"` (from memory).

File: boot/makebootable.go (hunk 5 above)
Comment (pedronis) [reply boot/makebootable.go]: changed to copyBootSnap

File: overlord/devicestate/handlers_install.go (hunk 3 above)
Comment (pedronis) [reply overlord/devicestate/handlers_install.go]: tweaked things around this

**Review by pedronis — APPROVED** [review]
+1 but I worked on it as well

**Review by mvo5 — COMMENTED** [review]
Fwiw, the changes from Samuele to this PR look fine to me.

File: boot/makebootable.go (hunk 2 above)
Comment (mvo5) [reply boot/makebootable.go:324]: I guess we could have used `filepath.EvalSymlinks()` but it should be fine, might be followup material.

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thank you. The change to use `RecoverySystemLabel` instead of `RecoverySystemDir` is great!

File: tests/lib/fakeinstaller/mk-classic-rootfs.sh [hunk 6]
```diff
@@ -80,8 +77,13 @@ if [ ! -d "$DST" ]; then
...
 fi
 
-# get the base
-wget -c http://cdimage.ubuntu.com/ubuntu-base/releases/22.04/release/ubuntu-base-22.04.1-base-amd64.tar.gz -O ubuntu-base.tar.gz
+# extract the base
+if [ -f /cdrom/casper/base.squashfs ]; then
+    sudo unsquashfs -d "$DESTDIR" /cdrom/casper/base.squashfs
+else
+    wget -c http://cdimage.ubuntu.com/ubuntu-base/releases/22.04/release/ubuntu-base-22.04.1-base-amd64.tar.gz -O ubuntu-base.tar.gz
+    sudo tar -C "$DESTDIR" -xf "$BASETAR"
```
Comment (alfonsosanchezbeato) [inline tests/lib/fakeinstaller/mk-classic-rootfs.sh:85]: I think $BASETAR is empty here?

**Pull request merged**